[nrf noup] Make sdk-connectedhomeip green again

Damian-Nordic opened the pull request:
Fetch cbor2 in bootstrap.sh
Cherry-pick a change to fix cirque CI
Update Python modules on nRF Connect update to harden things in the future.
Make build of nRF Connect examples optional (continue-on-error).

Signed-off-by: Damian Krolik <damian.krolik@nordicsemi.no>

kkasperczyk-no commented:
Please cherry-pick also this one: https://github.com/project-chip/connectedhomeip/commit/3ef748cf8ac11031becee1550536a43e1331de63, as in NCS update commit I introduced a bug that was fixed later on.

Damian-Nordic commented:
> Please cherry-pick also this one: [project-chip/connectedhomeip@3ef748c](https://github.com/project-chip/connectedhomeip/commit/3ef748cf8ac11031becee1550536a43e1331de63), as in NCS update commit I introduced a bug that was fixed later on.

Ok, I did and squashed with the previous commit.
